berkdb/mp/mp_fopen.c

```diff
@@ -1440,6 +1424,17 @@ __memp_mf_discard(dbmp, mfp)
 	/* Delete from the list of MPOOLFILEs. */
 	R_LOCK(dbenv, dbmp->reginfo);
 	SH_TAILQ_REMOVE(&mp->mpfq, mfp, q, __mpoolfile);
+	if ((fileid_mpf = hash_find(mp->mpfhash, mfp->fileid)) != NULL) {
+		LISTC_FOR_EACH_SAFE(&fileid_mpf->mpflist, fnd, tmp, lnk) {
+			if (fnd == mfp) {
+				listc_rfl(&fileid_mpf->mpflist, fnd);
+				if (listc_size(&fileid_mpf->mpflist) == 0) {
+					hash_del(mp->mpfhash, fileid_mpf);
+					free(fileid_mpf);
+				}
+			}
```

Comment:
break here to stop the loop?

Reply:
sure but the list will be empty at that point ..

Reply:
oh I think its a bug without the break .. wow .. thanks ..